src/aspire/nfft/finufft.py

```diff
@@ -1,33 +1,60 @@
-import finufftpy
 import numpy as np
-
+import finufftpy
 from aspire.nfft import Plan
 from aspire.utils import ensure
 
 
 class FINufftPlan(Plan):
-    def __init__(self, sz, fourier_pts, epsilon=1e-15, **kwargs):
+    def __init__(self, sz, fourier_pts, epsilon=1e-15, many=0, **kwargs):
         """
         A plan for non-uniform FFT (3D)
+
         :param sz: A tuple indicating the geometry of the signal
         :param fourier_pts: The points in Fourier space where the Fourier transform is to be calculated,
             arranged as a 3-by-K array. These need to be in the range [-pi, pi] in each dimension.
         :param epsilon: The desired precision of the NUFFT
+        :param many: Optional integer indicating if you would like to compute a batch of `many`
+        transforms.  Implies vol_f.shape is (..., `many`). Defaults to 0 which disables batching.
         """
+
+        self.many = False
+        self.ntransforms = 1
+        manystr = ''
+        if many != 0:
+            self.many = True
+            self.ntransforms = many
+            manystr = 'many'
         self.sz = sz
         self.dim = len(sz)
+
+        # TODO: Currently finufftpy is hardcoded as doubles only right now
+        self.dtype = fourier_pts.dtype
+        assert self.dtype == np.float64, 'finufftpy is hardcoded to doubles.'
```

Comment:
I'm okay with not raising here because I know finufft's pybind is going to upcast these on copy, but we'd need to ensure downcast on return to keep type's consistent.

Finufftpy can support both dtypes, it just doesn't atm... might be a little bit of work to do that.

Reply:
Noting that the Guru V2 branch work should result in an FINUFFT package that can support both types when it (and the python package updates) makes it into master.

Reply:
I have decided to be more flexible here to avoid the conflicts that would ensue.  It will be corrected when the latest host FINUFFT package is integrated and made a requirement.